### PR TITLE
Requests and Responses, Route Parameters

### PR DESCRIPTION
**Pull request by droberts-ada**
## Description
- Fill out the lesson on requests and responses
- Add a lesson on route parameters (how did we not have this?)

Ugh this is coming in much hotter than I would have liked. But, again it's just me writing down what we've said out loud every cohort.

Route parameters are _kind of_ talked about in the restful routing resource (which has been moved to Wednesday), but just barely. This is a consistent sticking point for students, so I feel it's worth having an explicit resource on it.

## Questions for the Author
- [x] Have you updated the weekly learning goals in the pedagogy resource
- [x] Have you added links to the source (lucidchart, draw.io) for any diagram

## Todos
Are any of these items still in progress?
- [ ] Update the calendar with the new route parameters lesson - target 30 minutes

### CODE REVIEW

**Comment by droberts-ada** [comment]
Ahhhhh sorry I'm asking for work over the weekend. If no one gets to this until Monday, well, we don't need it until the afternoon.

File: 08-rails/requests-and-responses.md [hunk 1]
```diff
@@ -33,46 +93,95 @@ In the table below, you can see the starter mapping between the HTTP Verb and th
...
+When added to a controller action, the `head` method indicates that only HTTP headers should be sent back (i.e. no payload). The only argument, the symbol `:not_found`, is the text-version of the status code to use (404).
+
+Typically you would do this inside of a conditional:
+
+```ruby
+def show
+  book_id = params[:id]
+  @book = BOOKS[book_id]
+  if @book.nil?
+    head :not_found
```
Comment (CheezItMan) [inline 08-rails/requests-and-responses.md]: Does `head` allow the rest of the page to render?  If not should you use `render` instead?

File: 08-rails/requests-and-responses.md [hunk 2]
```diff
@@ -7,9 +7,68 @@
...
+1. **Client** receives the response and displays it to the user (rendering the page)
+1. **User** takes another action, and the cycle begins anew
+
+We can shorten this to 3 major steps:
+1. Request
+1. Processing
+1. Response
+
+![Request response overview](images/request_response_overview.png)
+<!-- https://www.draw.io/#G17ynKwAoksm5XSU5tqEsDiGuMz_UwZByl -->
```
Comment (kariabancroft) [inline 08-rails/requests-and-responses.md]: Thanks for including this link!

File: 08-rails/requests-and-responses.md [hunk 3]
```diff
@@ -33,46 +93,95 @@ In the table below, you can see the starter mapping between the HTTP Verb and th
...
-## CRUD
-<b>C</b>reate. <b>R</b>ead. <b>U</b>pdate. <b>D</b>elete. This acronym embodies a huge majority of what we do with our web applications both simple and complex. We begin architecting our applications routes and interfaces by considering what kind of actions users will perform (CRUD) and what effects those actions will have on resources within the application (idempotency).
+Again, we can see there's a bunch of other HTTP stuff in there that we're ignoring for right now.
 
-## Summary
+### Status Codes
+
+Status codes are an important part of the HTTP protocol - they let the client know how things went. Every response must have a status code attached.
+
+Each status code is a 3-digit number paired with an English word or phrase. The first digit of the number indicates what sort of status it is, and the last two tell the details. It's sort of like how the first two digits of a room number are the floor that room is on.
```
Comment (kariabancroft) [inline 08-rails/requests-and-responses.md]: Yes! Love this
